ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,203 @@
+import { hot } from 'react-hot-loader';
+import React, { Component } from 'react';
+import { bool } from 'prop-types';
+import TextField from '@material-ui/core/TextField/TextField';
+import FormGroup from '@material-ui/core/FormGroup/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel/FormControlLabel';
+import FormLabel from '@material-ui/core/FormLabel/FormLabel';
+import MenuItem from '@material-ui/core/MenuItem/MenuItem';
+import CodeEditor from '@mozilla-frontend-infra/components/CodeEditor';
+import ListItem from '@material-ui/core/ListItem/ListItem';
+import Switch from '@material-ui/core/Switch/Switch';
+import { withStyles } from '@material-ui/core';
+import CheckIcon from 'mdi-react/CheckIcon';
+import List from '../../Documentation/components/List';
+import isWorkerTypeNameValid from '../../../utils/isWorkerTypeNameValid';
+import Dashboard from '../../../components/Dashboard';
+import Button from '../../../components/Button';
+
+const gcpConfig = {
+  minCapacity: 0,
+  maxCapacity: 0,
+  capacityPerInstance: 1,
+  machineType: 'n1-highcpu-8',
+  regions: ['us-west2'],
+  userData: {},
+  scheduling: {},
+  networkInterfaces: [{}],
+  disks: [{}],
+};
+
+@hot(module)
+@withStyles(theme => ({
+  successIcon: {
+    ...theme.mixins.successIcon,
+  },
+  createIconSpan: {
+    ...theme.mixins.fab,
+    ...theme.mixins.actionButton,
+    right: theme.spacing.unit * 11,
```

Comment:
The plan is to have also `Cancel` button for both creation and editing UIs, and `Delete` button for the editing variant

Reply:
> The plan is to have also Cancel button for both creation

The user could always use the browser navigator back button to cancel. At least this is how we do it in the rest of the app. We use cancel buttons for dialog since it's a little bit harder to close with the browser navigation.

> and Delete button for the editing variant <----- actually, a bad idea

I think that's a great idea :) We would still need a delete button if the user navigates to this worker pool directly via the URL.

Reply:
> The user could always use the browser navigator back button to cancel

Nowadays people are used to SPAs, where it's undesirable to navigate around using browser buttons, because that might mess up the app's state. Personally, on our Tools, I find the `Cancel` button lacking. It always takes me several seconds to look around for a `Cancel` button, then stop, remember that in this case it's ok to navigate with the browser buttons and then press the browser button. Why this complication when we can just have a `Cancel` button? Again, I volunteer to add everywhere for consistency

> We would still need a delete button if the user navigates to this worker pool directly via the URL.

oh that's fair. Why don't I add `Delete` buttons to both the editor and the table then! (not in this PR tho)